### ISSUE TITLE
[Bug] FluentSlider two-way binding issue

### ISSUE THREAD

**Issue by LuohuaRain** [microsoft/fluentui-blazor#2609]
https://github.com/microsoft/fluentui-blazor/pull/2591#issuecomment-2324344948

While the mouse wheel is just one example, there are other ways to interact with the slider. For instance, how can I reset the slider value? Fluent UI React Web Component supports this functionality.

https://react.fluentui.dev/?path=/docs/components-slider--docs#controlled

```jsx
import * as React from "react";
import { useId, Button, Label, Slider } from "@fluentui/react-components";
import type { SliderProps } from "@fluentui/react-components";

export const Controlled = () => {
  const id = useId();
  const [sliderValue, setSliderValue] = React.useState(160);
  const onSliderChange: SliderProps["onChange"] = (_, data) =>
    setSliderValue(data.value);
  const resetSlider = () => setSliderValue(0);
  return (
    <>
      <Label htmlFor={id}>
        Control Slider [ Current Value: {sliderValue} ]
      </Label>
      <Slider
        aria-valuetext={`Value is ${sliderValue}`}
        value={sliderValue}
        min={20}
        max={200}
        onChange={onSliderChange}
        id={id}
      />

      <Button onClick={resetSlider}>Reset</Button>
    </>
  );
};

```

React can do this, so how blazor do this?

```razor
<div style="display: flex; flex-direction: column; align-items: flex-start; height: min-content;">
    <label for="rtl-slider" style="margin-left: 8px;">rtl support</label>
    <div dir="rtl" style="width: 100%;">
        <FluentSlider id="rtl-slider" Min="0" Max="100" Step="10" @bind-Value=value2>
            <FluentSliderLabel Position="10">10</FluentSliderLabel>
            <FluentSliderLabel Position="20">20</FluentSliderLabel>
            <FluentSliderLabel Position="40">40</FluentSliderLabel>
            <FluentSliderLabel Position="60">60</FluentSliderLabel>
            <FluentSliderLabel Position="80">80</FluentSliderLabel>
        </FluentSlider>
        
        <FluentButton OnClick="() => value2 = 0 ">Reset</FluentButton>  @* THIS LINE IMPORTANT *@
    </div>
</div>
```

## 🤔 Expected Behavior
I change the value by clicking/dragging, and after I want to reset, then I click `Reset` and the FluentSlider Value should be 0. **That's what true two-way data binding means**

## 😯 Current Behavior
Currently, the value does not reset.

**Comment by vnbaaij** [comment]
The fact that the React component does that, says absolutely nothing about what the we component can do. I think we have an example somewhere. I'll get back on that

**Comment by LuohuaRain** [comment]
> The fact that the React component does that, says absolutely nothing about what the web component can do. 

Yes sure!

The example with the React component isn't quite relevant since it doesn't appear to be a Web Component. However, the FAST Element can change the Slider's position by modifying the value property. Wouldn't this example be more convincing?

```html
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <script type="module"
        src="https://cdn.jsdelivr.net/npm/@microsoft/fast-components/dist/fast-components.min.js"></script>
    <title>Document</title>
</head>

<body>
    <fast-slider id="slider" min="0" max="100" step="10" value="70">
        <fast-slider-label position="0">
            0
        </fast-slider-label>
        <fast-slider-label position="10">
            10
        </fast-slider-label>
        <fast-slider-label position="90">
            90
        </fast-slider-label>
        <fast-slider-label position="100">
            100
        </fast-slider-label>
    </fast-slider>
    <fast-button id="resetButton">Reset</fast-button>

    <script>
        document.getElementById('resetButton').addEventListener('click', function () {
            document.getElementById('slider').value = 0; 
        });
    </script>
</body>

</html>
```

**Comment by oneolddev** [comment]
@vnbaaij 

This issue looks familiar - I submitted #1873 that fixed a redraw issue #1873. It looks like #2287 removed some of that code and may be the reason why this issue is now occurring. 

It's been a while but I recall something about how the underlying webcomponent initializes. I can take a look in the morning if you wish.

**Comment by vnbaaij** [comment]
Yes, it is familiar but it also isn't :)

What I've found so far:
- Addding below to `FluentSlider.razor`
```
current-value="@FormatValueAsString(Value)"  @* ADD THIS LINE *@
aria-valuenow="@FormatValueAsString(Value)" @* ADD THIS LINE *@
```

Works when using the Blazor code from @LuohuaRain's opening post. What does not work is using the Slider inside another component that can influence the slider's value. I'm using the DataGrid Typical Usage filter example for this. moving the thumb a couple of times in quick succession throws it into a loop:
![grid-slider](https://github.com/user-attachments/assets/559e8432-ee18-4f59-a81d-5f27d3c67a30)

Not adding the `current-value` to the slider tag and using the script for earlier PR that Gary (@oneolddev) mentioned has the same result (reset works, grid goes into loop) **if we invoke the js function on every render** (ie in `OnAfterRenderAsync` but outside of `firstRender' block):
```
protected override async Task OnAfterRenderAsync(bool firstRender)
{
    if (firstRender)
    {
        _jsModule ??= await JSRuntime.InvokeAsync<IJSObjectReference>("import", JAVASCRIPT_FILE);
    }
    if (_jsModule is not null)
    {
        await _jsModule!.InvokeVoidAsync("updateSlider", Element);
    }
}
```

If we move the script call to inside the `firstRender` block, the data grip filter example works flawlessly. @LuohuaRain's code works for some part not not completely. If you move a slider and then press reset, the value gets set to 0 but the thumb is not updated. If you press reset without moving a slider, it will set the value and the thumb correctly:
![issue-slider](https://github.com/user-attachments/assets/de4da29d-c3a3-4ad1-935f-4cf72c3b5505)

Now we just need a working combination of these two. I've tried everything I could think of but was not able to find a definitive solution.  So Gary, if your offer still stands, please take a look.

**Comment by oneolddev** [comment]
I'm looking into this now. I was not aware of the datagrid and slider interaction. ⛏️ ⛏️

**Comment by vnbaaij** [comment]
Closed with PR #2665
